include/picongpu/simulation/stage/Collision.hpp

```diff
@@ -47,19 +67,135 @@ namespace picongpu
                 {
                 }
 
+                //! Calculates the Debye length squared from the sum of the species contributions
+                struct GetSquaredScreeningLength
+                {
+                    template<typename T_Acc>
+                    DINLINE void operator()(T_Acc const& acc, float1_X& lengthInvSquared, const float1_X& dens) const
+                    {
+                        /* avoid dividing by zero, lengthInvSquared is zero if there were no charged particles.
+                         * In that case the Debye length should also be set to zero.
+                         */
+                        if(lengthInvSquared[0] <= std::numeric_limits<float_X>::min())
+                        {
+                            // if no particles set debye length to zero
+                            lengthInvSquared[0] = 0.0_X;
+                            return;
+                        }
+                        // the default case
+                        const float_X squaredLength = 1.0_X / lengthInvSquared[0];
+
+                        // enforce a lower cutoff for the Debye length equal to the mean inter atomic distance
+                        // of the species with the highest density
+                        const float_X maxDens = dens[0] * particles::TYPICAL_NUM_PARTICLES_PER_MACROPARTICLE;
+                        const float_X val = 2.0_X * pmacc::math::Pi<float_X>::doubleValue / 3.0_X * maxDens;
+                        const float_X rMin = 1.0_X / math::cbrt(val);
+                        const float_X rMin2 = rMin * rMin;
+                        lengthInvSquared[0] = math::max(squaredLength, rMin2);
+                    }
+                };
                 /** Perform particle particle collision
                  *
                  * @param step index of time iteration
                  */
-                void operator()(uint32_t const step) const
+                void operator()(uint32_t const currentStep) const
                 {
-                    pmacc::meta::ForEach<
-                        particles::collision::CollisionPipeline,
-                        particles::collision::CallCollider<bmpl::_1>>{}(m_heap, step);
+                    // Calculate squared Debye length using the formula form [Perez 2012].
+                    // A species temperature is assumed to be (2/3)<E_kin>
+                    if constexpr(particles::collision::calculateScreeningLength)
```

Comment:
Could you please move the full body including the if condition into a member method `debug` to make this method more readable?
This would visible separate the debug part from the real code usage.